xline/src/server/kv_server.rs

```diff
@@ -195,21 +196,62 @@ where
     }
 
     /// Validate range request before handle
-    fn check_range_request(req: &RangeRequest, current_revision: i64) -> Result<(), tonic::Status> {
+    fn check_range_request(
+        req: &RangeRequest,
+        compacted_revision: i64,
+        current_revision: i64,
+    ) -> Result<(), tonic::Status> {
         if req.key.is_empty() {
             return Err(tonic::Status::invalid_argument("key is not provided"));
         }
         if !SortOrder::is_valid(req.sort_order) || !SortTarget::is_valid(req.sort_target) {
             return Err(tonic::Status::invalid_argument("invalid sort option"));
         }
         if req.revision > current_revision {
-            return Err(tonic::Status::invalid_argument(format!(
+            Err(tonic::Status::invalid_argument(format!(
                 "required revision {} is higher than current revision {}",
                 req.revision, current_revision
-            )));
+            )))
+        } else {
+            Self::check_range_compacted(req.revision, compacted_revision)
         }
+    }
 
-        Ok(())
+    /// check whether the required revision is compacted or not
+    fn check_range_compacted(
+        range_revision: i64,
+        compacted_revision: i64,
+    ) -> Result<(), tonic::Status> {
+        (range_revision >= compacted_revision)
```

Comment:
The range_revision cound be <=0, when means use the latest revision, we should return Ok.

Reply:
Done